test(cadl): initial sdk for adp

archerzz opened the pull request:
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

archerzz committed:
test(cadl): initial sdk for adp

archerzz committed:
move csproj

archerzz committed:
update csproj

archerzz committed:
fix csproj name

archerzz committed:
fix compilation error

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.AutonomousDevelopmentPlatform](https://apiview.dev/Assemblies/Review/3e31162ebda9446da9f23bf8c5620cb1)

archerzz committed:
update using template

archerzz committed:
pull in shared codes

archerzz committed:
remove unimplemented link

m-nash commented on sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/GlobalSuppressions.cs:
```diff
@@ -0,0 +1,6 @@
+
+using System.Diagnostics.CodeAnalysis;
+
+[assembly: SuppressMessage("Usage", "AZC0002:DO ensure all service methods, both asynchronous and synchronous, take an optional CancellationToken parameter called cancellationToken.", Justification = "CancellationToken can be passed through RequestOptions")]
```
FYI https://github.com/Azure/azure-sdk-tools/issues/4715

archerzz replied:
@m-nash I think the global suppression here is unnecessary. I've posted details in the issue above.

m-nash commented on sdk/adp/Azure.AI.AutonomousDevelopmentPlatform/src/properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
```
This folder should be capital `Properties`

archerzz replied:
done

archerzz committed:
rename folder `properties` to `Properties`

archerzz committed:
remove global suppression which is unnecessary

archerzz commented:
corresponding API view: https://apiview.dev/Assemblies/Review/f0323a0a0e7b4679896fd38c2baf6ec6

archerzz committed:
fix error MSB5029

After the MSBuild version was upgraded from 17.0.0+c9eb9dd64 to 17.3.2+561848881, there is a new warning MSB5029 complaining that `/**/*.cs` should be avoided.
The fix is to remove `/**`. We should be able to remove the whole `Include` after we fix the CADL library build in `azure-sdk-for-net`.
